Packs/MicrosoftECM/Integrations/MicrosoftECM/MicrosoftECM.ps1

```diff
@@ -421,6 +421,12 @@ Function ExecuteServiceScript()
 		param($SiteCode, $CollectionID, $CollectionName, $DeviceName, $ScriptText, $ScriptName)
 		Set-Location $env:SMS_ADMIN_UI_PATH\..\
 		Import-Module .\ConfigurationManager.psd1
+
+		# Connect to the site's drive if it is not already present
+        if((Get-PSDrive -Name $SiteCode -PSProvider CMSite -ErrorAction SilentlyContinue) -eq $null) {
+            New-PSDrive -Name $SiteCode -PSProvider CMSite -Root $computerName -Verbose $true -ErrorAction "Stop"
```

Comment:
We can have this logic in a function `Function HandlePSDrive()` and call it from everywhere we need it